python/runtime/common/py_NoiseModel.cpp

```diff
@@ -221,6 +246,27 @@ void bindNoiseChannels(py::module &mod) {
       .def(
           py::init<double>(), py::arg("probability"),
           "Initialize the `PhaseFlipChannel` with the provided `probability`.");
+
+  py::class_<readout_error, kraus_channel>(
+    mod, "ReadoutError",
+    R"#(Models the probabilistic errors that can happen when measuring qubits.
+    It allows to define a custom probability distribution that describes how
+    often a qubit's measurement result differs from its actual state.
+    Its constructor expects a 2x2 matrix of float values that represents the
+    probability matrix applied to the probabilistic state of the qubit measurement.
+    It takes the form of
+
+    P = [[p(0|0), p(0|1)], [p(1|0), p(1,1)]]
+
+    where p(i|j) is the probability of measuring outcome i when the actual state is j
+
+    )#")
+    .def(py::init([](const py::buffer *b) {
+      py::buffer_info info = b->request();
+      std::vector<real> v(info.shape[0] * info.shape[1]);
```

Comment:
Shouldn't we make the allocation in extractRealData instead of here? We use memcopy inside the extractRealData function and we assume that we have enough allocated memory. As it happens we have because we only support real types for now. Maybe we could pass a vector reference instead of a pointer for now and use templates in the future if we need to support multiple data types.

Reply:
you are absolutely right, the only reason I did it like this since it follows a pattern of other pyBind exposed methods,